src/smtp_server.c

```diff
@@ -642,8 +662,57 @@ int smtp_hdlr_data(struct smtp_server_context *ctx, const char *cmd, const char
 	/* prepare response */
 	ctx->code = 354;
 	ctx->message = strdup("Go ahead");
-	ctx->node = smtp_cmd_lookup("BODY");
-	return SCHS_CHAIN;
+
+	smtp_server_response(stream, ctx->code, ctx->message);
+
+	// Parse the BODY content of DATA
+	struct im_header_context im_hdr_ctx = IM_HEADER_CONTEXT_INITIALIZER;
+	struct stat stat;
+
+	assert_mod_log(ctx->body.stream != NULL);
+
+	im_hdr_ctx.max_size = 65536; // FIXME use proper value
+	im_hdr_ctx.hdrs = &ctx->hdrs;
+	//sleep(10);
+	switch (smtp_copy_to_file(ctx->body.stream, stream, &im_hdr_ctx)) {
+		case 0:
+			insert_received_hdr(ctx);
+			ctx->code = 250;
```

Comment:
Yes. It will be overwritten by the message received from JS - unless there is not sufficient storage space (see the check below the switch), and in that case a different message will be generated anyway.